pdf/pdflinkarea.cpp

```diff
@@ -66,23 +73,40 @@ void PDFLinkArea::mousePressEvent(QMouseEvent *event)
 {
     d->clickInProgress = true;
     d->clickLocation = event->pos();
+    d->pressTimer.start();
 }
 
-void PDFLinkArea::mouseReleaseEvent(QMouseEvent *event)
+void PDFLinkArea::mouseMoveEvent(QMouseEvent *event)
 {
+    // Don't activate anything if the finger has moved too far
+    QRect rect((d->clickLocation - QPointF(d->wiggleFactor, d->wiggleFactor)).toPoint(),
+               QSize(d->wiggleFactor * 2, d->wiggleFactor * 2));
+    if (!rect.contains(event->pos()))
+        return;
     d->clickInProgress = false;
```

Comment:
ClickInProgress seems to be in practice the same thing as timer running. Could perhaps simplify code by just using timer state?

As a detail here the timer is left running but the handler checks clickInProgress.